lib/ansible/modules/network/cli/cli_config.py

```diff
@@ -0,0 +1,291 @@
+  returned: always
+  type: list
+  sample: ['interface Loopback999', 'no shutdown']
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import Connection
+from ansible.module_utils._text import to_text
+
+
+def validate_args(module, capabilities):
+    """validate param if it is supported on the platform
+    """
+    if (module.params['replace'] and
+            not capabilities['device_operations']['supports_replace']):
+        module.fail_json(msg='replace is not supported on this platform')
+
+    if (module.params['rollback'] and
+            not capabilities['device_operations']['supports_rollback']):
+        module.fail_json(msg='rollback is not supported on this platform')
+
+    if (module.params['commit_comment'] and
+            not capabilities['device_operations']['supports_commit_comment']):
+        module.fail_json(msg='commit_comment is not supported on this platform')
+
+    if (module.params['defaults'] and
+            not capabilities['device_operations']['supports_defaults']):
+        module.fail_json(msg='defaults is not supported on this platform')
+
+    if (module.params['multiline_delimiter'] and
+            not capabilities['device_operations']['supports_multiline_delimiter']):
+        module.fail_json(msg='multiline_delimiter is not supported on this platform')
+
+    if (module.params['diff_replace'] and
+            not capabilities['device_operations']['supports_diff_replace']):
+        module.fail_json(msg='diff_replace is not supported on this platform')
+
+    if (module.params['diff_match'] and
+            not capabilities['device_operations']['supports_diff_match']):
+        module.fail_json(msg='diff_match is not supported on this platform')
+
+    if (module.params['diff_ignore_lines'] and
+            not capabilities['device_operations']['supports_diff_ignore_lines']):
+        module.fail_json(msg='diff_ignore_lines is not supported on this platform')
+
+
+def run(module, connection, candidate, running):
+    result = {}
+
+    replace = module.params['replace']
+    rollback = module.params['rollback']
+    commit_comment = module.params['commit_comment']
+    multiline_delimiter = module.params['multiline_delimiter']
+    diff_replace = module.params['diff_replace']
+    diff_match = module.params['diff_match']
+    diff_ignore_lines = module.params['diff_ignore_lines']
+
+    commit = not module.check_mode
+
+    if replace in ('yes', 'true', 'True'):
+        replace = True
+    elif replace in ('no', 'false', 'False'):
+        replace = False
+
+    kwargs = {'candidate': candidate, 'running': running}
+    if diff_match:
+        kwargs.update({'diff_match': diff_match})
+    if diff_replace:
+        kwargs.update({'diff_replace': diff_replace})
+    if diff_ignore_lines:
+        kwargs.update({'diff_ignore_lines': diff_ignore_lines})
+
+    diff_response = connection.get_diff(**kwargs)
```

Comment:
get_diff() need to be called only when device doesn't support on-box diff ie. check for value of `supports_onbox_diff` in device capability.

Reply:
If device supports onbox diff directly call edit_config() with required parameters the diff part will be taken care by device and if diff is present will be added in the edit_config() resposne